Crash when loading activity that extends BaseActivity

Here is the error
```error
java.lang.NullPointerException: null cannot be cast to non-null type androidx.navigation.fragment.NavHostFragment
  at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6032)
  at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:6077)
  at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:47)
  at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
  at android.os.Handler.dispatchMessage(Handler.java:106)
  at android.os.Looper.loopOnce(Looper.java:226)
  at android.os.Looper.loop(Looper.java:313)
  at android.app.ActivityThread.main(ActivityThread.java:8751)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
  Caused by: java.lang.NullPointerException: null cannot be cast to non-null type androidx.navigation.fragment.NavHostFragment
  at dev.teogor.ceres.extensions.ActivityExtKt.findNavController(ActivityExt.kt:28)
  at dev.teogor.ceres.components.app.BaseActivity.findNavController(BaseActivity.kt:262)
  at dev.teogor.ceres.components.app.BaseActivity.onDestroy(BaseActivity.kt:181)
  at android.app.Activity.performDestroy(Activity.java:8571)
  at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1364)
  at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6019)
  at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:6077) 
  at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:47) 
  at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
  at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443) 
  at android.os.Handler.dispatchMessage(Handler.java:106) 
  at android.os.Looper.loopOnce(Looper.java:226) 
  at android.os.Looper.loop(Looper.java:313) 
  at android.app.ActivityThread.main(ActivityThread.java:8751) 
  at java.lang.reflect.Method.invoke(Native Method) 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 
```

--- a/ceres-extensions/api/ceres-extensions.api
+++ b/ceres-extensions/api/ceres-extensions.api
@@ -6,6 +6,10 @@ public final class dev/teogor/ceres/extensions/ActivityExtKt {
 	public static synthetic fun hasIntentExtras$default (Landroid/app/Activity;Lkotlin/jvm/functions/Function1;ILjava/lang/Object;)V
 }
 
+public final class dev/teogor/ceres/extensions/AnyExt {
+	public static final fun safeReturn (Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
+}
+
 public final class dev/teogor/ceres/extensions/ApplicationExtKt {
 	public static final fun activityLifecycle (Landroid/app/Application;Lkotlin/jvm/functions/Function2;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;Lkotlin/jvm/functions/Function1;)Landroid/app/Application$ActivityLifecycleCallbacks;
 	public static synthetic fun activityLifecycle$default (Landroid/app/Application;Lkotlin/jvm/functions/Function2;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;Lkotlin/jvm/functions/Function1;ILjava/lang/Object;)Landroid/app/Application$ActivityLifecycleCallbacks;

--- a/ceres-extensions/src/main/kotlin/dev/teogor/ceres/extensions/ActivityExt.kt
+++ b/ceres-extensions/src/main/kotlin/dev/teogor/ceres/extensions/ActivityExt.kt
@@ -27,8 +27,11 @@ fun FragmentActivity.findNavController(
   @IdRes viewId: Int
 ): NavController? {
   return if (viewId.validIdRes) {
-    val navHostFragment = this.supportFragmentManager.findFragmentById(viewId) as NavHostFragment
-    navHostFragment.navController
+    val fragment = this.supportFragmentManager.findFragmentById(viewId)
+    fragment.safeReturn {
+      val navHostFragment = this as NavHostFragment
+      navHostFragment.navController
+    }
   } else {
     null
   }

--- a/ceres-extensions/src/main/kotlin/dev/teogor/ceres/extensions/AnyExt.kt
+++ b/ceres-extensions/src/main/kotlin/dev/teogor/ceres/extensions/AnyExt.kt
@@ -1,0 +1,29 @@
+/*
+ * Copyright 2022 teogor (Teodor Grigor) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("AnyExt")
+
+package dev.teogor.ceres.extensions
+
+fun <Type, ReturnType> Type?.safeReturn(
+  nonNullRun: Type.() -> ReturnType
+): ReturnType? {
+  return if (this == null) {
+    null
+  } else {
+    nonNullRun()
+  }
+}

